src/sinks/loki/config.rs

```diff
@@ -36,6 +36,34 @@ impl EncodingConfigMigrator for EncodingMigrator {
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum ExtendedCompression {
+    #[serde(rename = "snappy")]
+    Snappy,
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
```

Comment:
```suggestion
/// Compression configuration for Loki.
#[configurable_component]
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Reply:
This needs a doc comment and to use `#[configurable_component]` as part of the on-going [configuration schema](https://github.com/vectordotdev/vector/issues/12141) work.

(This will make more sense once you merge in the latest `master` and see how the `config.rs` has since changed.)